client/components/domains/register-domain-step/index.jsx

```diff
@@ -337,9 +337,26 @@ class RegisterDomainStep extends Component {
 		return strippedHostname ?? hostname;
 	}
 
+	getInitialQueryInAIFlow() {
```

Comment:
Maybe make this generic for better usability? Something like `getInitialQueryFromSiteName`. You can keep it only active for your flow but then people could add their flows as well.

Reply:
https://github.com/Automattic/wp-calypso/pull/100948/commits/82ef7eb326213c02f38296ab314a690ba9e8b7cf